02-sql-aggregation.md

```diff
@@ -114,32 +114,121 @@ think of views is as a table, that can read, aggregate, and filter information
-before the query itself. For example, if we want to save the query giving
-the number of individuals in a view, we can write
+before the query itself. For example, imagine that my project only covers
+the data gathered during the summer (May - September) of 2000.  That 
+query would look like: 
 
-    CREATE VIEW species_count AS
-    SELECT species_id, COUNT(*)
+    SELECT *
     FROM surveys
-    GROUP BY species_id;
+    WHERE year == 2000 AND (month > 4 AND month < 10)
 
-Now, we will be able to access these results with a much shorter notation:
+But we don't want to have to type that every time we want to ask a 
+question about that particular subset of data!  Let's create a view: 
 
+    CREATE VIEW summer_2000 AS
     SELECT *
-    FROM species_count;
-
-Assuming we do not need this view anymore, we can remove it from the database
-almost as we would a table:
-
-    DROP VIEW species_count;
+    FROM surveys
+    WHERE year == 2000 AND (month > 4 AND month < 10)
 
 You can also add a view using *Create View* in the *View* menu and see the
 results in the *Views* tab just like a table
 
+Now, we will be able to access these results with a much shorter notation:
+
+    SELECT *
+    FROM summer_2000;
+
 > ### Challenge
 >
 > Write a query that returns the number of each species
 caught in each year sorted from most often caught species to the least
 occurring ones within each year starting from the most recent records. Save
 this query as a `VIEW`.
 
+## Null values
+
+Using the view we created in the previous section (`summer_2000`), let's 
+talk about null values.  Missing values in SQL are identified with the 
+special NULL value.  Scroll through our `summer_2000` view.  It should be 
+easy to find several records with missing values.  How do you think we 
+could filter to find these rows?  
+
+You might want to try a query like: 
+
+    SELECT *
+    FROM summer_2000
+    WHERE species_id == NULL
+
+But this won't work, because the NULL value isn't a text string, it's 
+a special value.  Instead we need the following syntax: 
+
+    SELECT *
+    FROM summer_2000
+    WHERE species_id IS NULL
+
+If we wanted to use all the records where `species_id` is NOT null, we 
+would add the `NOT` keyword to our query.  
+
+    SELECT *
+    FROM summer_2000
+    WHERE species_id IS NOT NULL
+
+There are many hidden "gotchas" with NULL values.  If we restrict our 
+query to the "PE" species, this will be easier to see: 
+
+    SELECT *
+    FROM summer_2000
+    WHERE species_id == 'PE'
+
+If we find the average weight, SQL behaves like we would hope, ignoring
+the NULL values:
+
+    SELECT AVG(weight)
+    FROM summer_2000
+    WHERE species_id == 'PE'
+
+But if we try to be extra clever, and find the average ourselves, 
+we might get tripped up: 
+
+   SELECT SUM(weight), COUNT(*), SUM(weight)/COUNT(*)
+   FROM summer_2000
+   WHERE species_id == 'PE'
+
+This *will* work if I modify the count command slightly: 
```

Comment:
This is difficult to understand for a first contact with SQL, maybe be more explicit about the issue and give only working examples.